genanki/__init__.py

```diff
@@ -155,25 +156,31 @@ class Card:
   def __init__(self, ord_):
     self.ord = ord_
 
-  def write_to_db(self, cursor, now_ts, deck_id, note_id):
+    self.interval = 0
+
+  def write_to_db(self, cursor, now_ts, deck_id, note_id,
+                  stage, queue, due, interval, ease, reps_til_grad):
     cursor.execute('INSERT INTO cards VALUES(null,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?);', (
-        note_id,    # nid
-        deck_id,    # did
-        self.ord,   # ord
-        now_ts,     # mod
-        -1,         # usn
-        0,          # type (=0 for non-Cloze)
-        0,          # queue
-        0,          # due
-        0,          # ivl
-        0,          # factor
-        0,          # reps
-        0,          # lapses
-        0,          # left
-        0,          # odue
-        0,          # odid
-        0,          # flags
-        "",         # data
+        note_id,    # nid - note ID
+        deck_id,    # did - deck ID
+        self.ord,   # ord - which card template it corresponds to
+        now_ts,     # mod - modification time as seconds since Unix epoch
+        -1,         # usn - value of -1 indicates need to push to server
+        stage,      # type - 0=new, 1=learning, 2=review
```

Comment:
Do you mean just within this function, or in the user-facing API as well?

Reply:
Actually, I remember now why I didn't use `type` for a variable name: it's a Python reserved keyword. Perhaps we should make an exception in this case?

Reply:
It's better to closely match Anki's naming. I want `genanki` to be a lower-level library that more-or-less directly maps to the way Anki represents data. Plus, the user will probably have to end up learning the Anki names anyway, for example if they're digging into the `collections.anki2` SQLite file to debug something. Abstractions are leaky more often than people expect.

I's OK to use `type` as the name. If you need to access the real `type()` function, you can import the `builtins` module and call `builtins.type`. (Previously I also had an instance of this, calling a param `ord_` instead of `ord`. But I changed it because it's silly to convolute your external API for the convenience of internal code, and I wasn't using `ord()` anyway).